design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+Also note that in the current implementation of policies, they must be in the same namespace as the claim referencing them.
+This will likely be updated in the future to support cluster or environment scoped policies.
+
+## Stacks
+
+Stacks are likely to represent higher level "application" functionality, but are not required to do so.
+Instead of providing the infrastructure and lower level platform components for entire environments like a cluster stack, they likely will merely provide the functionality to support a single application.
+It is expected for stacks to (indirectly) consume and build on functionality from cluster stacks in the form of resource claims.
+For example, a stack could generate a `MySQLInstance` claim that is fulfilled through a default resource class by a cluster stack in the form of a concrete MySQL database instance.
+
+The key aspect of a stack is that it will be installed with permissions to only watch for resources **within the namespace** it is running in.
+This provides a nice isolation as well as versioning story.
+
+* **Isolation:** The stack should not be able to read or write any resources outside of its target namespace.
+* **Versioning:** Multiple versions of the stack can be installed within the control plane, each namespace can have their own independent version.
+
+### Installation Flow
+
+Let's look at a concrete example of what happens when the `WordPress` stack is installed into Crossplane in a target namespace of the user's choice:
+
+* The user creates an `StackInstall` and specifies the WordPress stack
+* All CRDs defined by the WordPress stack will be installed into the cluster (remember that CRDs themselves are always cluster scoped).
+In the case of WordPress, a single CRD is installed that represents a WordPress application instance, `WordPressInstance`.
+* A `Role` will be created in the target namespace that includes the RBAC rules for the [allowed core Kubernetes kinds](#allowed-resource-access), the stack's owned CRDs, and its dependent CRDs.
+Because this is a `Role`, this only grants the given permissions to namespaced resources in the target namespace.
+* A `ServiceAccount` is created in the namespace that the stack's controller will be running that will serve as the runtime identity of the controller.
+* A `RoleBinding` is created in the target namespace to bind the namespaced `Role` and the namespaced `ServiceAccount`.
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the WordPress controller.
+
+### Architecture Diagram - Stack
+
+The diagram below summarizes the artifacts in the control plane after the WordPress stack has been installed and an example `WordPressInstance` custom resource has been created.
+The diagram shows the result of the following sequence of actions:
+
+1. The user creates an `StackInstall` and specifies the WordPress stack
```

Comment:
Nitpick: `an` should be `a`

Reply:
will update this, as well as the other instance of this typo above